server/controllers/budget.js

```diff
@@ -0,0 +1,125 @@
+/* Budget controller */
+'use strict';
+
+var csv    = require('fast-csv'),
+	fs     = require('fs'),
+	q      = require('q');
+
+var config = require('../config/environment/server'),
+	db     = require('../lib/db');
+	
+var uploadDir = config.uploadFolder;
+
+module.exports = {
+	update : updateBudget,
+	upload : uploadedFile
+};
+
+function uploadedFile(req, res, next) {
+	/*
+		* Objectif : handling the uploaded file and generate an array (csvArray) from the file
+	*/
+	var csvArray = [];
+	var file = req.files.file;
+	var fiscal_year_id = req.body.fiscal_year_id;
+	var period = JSON.parse(req.body.period);
+
+	if (file && file.path && file.name) {
+		var fileName = uploadDir + file.name.replace(/ /g, '_');
+
+		// File name can be more readable
+		fs.rename(file.path, fileName, function (err) {
+			if (err) { throw err; }
+			fs.stat(fileName, function (err, stat) {
+				if (err) { throw err; }
+				csv.fromPath(fileName, {headers: true})
+				.on('data', function (data) {
+					csvArray.push(data);
+				})
+				.on('end', function () {
+					createBudget(csvArray, fiscal_year_id, period);
+					res.status(200).send('uploaded');
+				});
+			});
+		});
+	} 
+}
+
+function createBudget(csvArray, fiscal_year_id, period) {
+
+	/*
+		* Objectif : create a new budget via params given
+		* Params : 
+		*	- csvArray : an array of the budget csv file
+		* - fiscal_year_id : the fiscal year id
+		* - period : the period
+	*/
+
+	if (csvArray.length) {
+
+		if (period.period_number === 0) {
+			// Budget for the fiscal year
+			var queryPeriods = 'SELECT period.id, period.period_number FROM period WHERE period.fiscal_year_id = ? ';
+			db.exec(queryPeriods, [fiscal_year_id])
```

Comment:
Where is the `catch()` statement for promise?  Is there any chance that an error could occur?